obsplus/bank/eventbank.py

```diff
@@ -273,9 +284,10 @@ def get_events(self, **kwargs) -> obspy.Catalog:
         {get_events_params}
         """
         paths = self.bank_path + self.read_index(columns="path", **kwargs).path
-        cats = (obspy.read_events(x) for x in paths)
+        read_func = partial(_try_read_catalog, format=self.format)
+        kwargs = dict(chunksize=len(paths) // CPU_COUNT)
```

Comment:
Instead of `CPU_COUNT` this could be `self.executor._max_workers` or this could be a class property - `self._max_workers` where this equal `self.executor._max_workers` if executor is not None, or 1 otherwise?

Reply:
This sounds reasonable, but I have two reservations:

1) `_max_workers` is a private attribute, so we need to keep a close eye on the API to make sure we correct it if it changes.
2) `_max_workers` is not part of the `Executor` interface which is problematic for the `dask.distributed.Executor`, for example, which doesn't have this attribute. 

Reading more into it, however, it appears the [dask executor doesn't honor the `map` return type of `concurrent.futures`](https://distributed.readthedocs.io/en/1.9.5/executor.html#map), nor does it allow for any batching through the use of `chunk_size`. Maybe we shouldn't try to support anything but `concurrent.futures` type `Executor`s? Of course, `Bank` instances can be used in other processes with more advanced concurrency, it just wouldn't be as convenient.